sdk/core/src/bytes_response.rs

```diff
@@ -0,0 +1,138 @@
+#[cfg(feature = "mock_transport_generate")]
+use crate::{collect_pinned_stream, BytesStream, Response, StreamError};
+use bytes::Bytes;
+use http::{HeaderMap, StatusCode};
+use serde::de::Visitor;
+use serde::ser::{Serialize, SerializeStruct, Serializer};
+use serde::{Deserialize, Deserializer};
+use std::collections::HashMap;
+
+const FIELDS: &'static [&'static str] = &["status", "headers", "body"];
+
+#[derive(Debug, Clone, PartialEq)]
+pub(crate) struct BytesResponse {
+    status: StatusCode,
+    headers: HeaderMap,
+    body: Bytes,
+}
+
+impl Serialize for BytesResponse {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut hm = HashMap::new();
+        for (h, v) in self.headers.iter() {
+            hm.insert(h.to_string(), v.to_str().unwrap());
+        }
+
+        let mut state = serializer.serialize_struct("Response", 3)?;
+        state.serialize_field(FIELDS[0], &self.status.as_u16())?;
+        state.serialize_field(FIELDS[1], &hm)?;
+        state.serialize_field(FIELDS[2], &self.body as &[u8])?;
+
+        state.end()
+    }
+}
+
+impl<'de> Deserialize<'de> for BytesResponse {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        deserializer.deserialize_struct("Response", FIELDS, ByteResponseVisitor)
+    }
+}
+
+struct ByteResponseVisitor;
+
+impl<'de> Visitor<'de> for ByteResponseVisitor {
```

Comment:
Why are we not deserializing using `serde_json` or just deriving `Deserialize` on the `BytesResponse` struct?

Reply:
Because `StatusCode`, `HeaderMap` and `Bytes` do not implement Deserialize. 🤷‍♂️ .

Reply:
I guess what I'm suggesting is having something like:

```rust
#[derive(Deserialize)]
struct SeriealizedBytesResponse {
  status: u16,
  headers: HashMap<String, String>
  body: Vec<u8>
}

impl std::convert::TryFrom<SeriealizedBytesResponse> for BytesResponse {
 // ...
}
```

This is bound to be a lot less code...